src/ansys/mapdl/core/cli.py

```diff
@@ -470,6 +470,10 @@ def start(
                 + " The following argument is not allowed in CLI: 'license_server_check'.\nIgnoring argument."
             )
 
+        # Ignoring env var if using CLI
+        if "PYMAPDL_START_INSTANCE" in os.environ:
+            os.environ.pop("PYMAPDL_START_INSTANCE")
+
```

Comment:
For now, `start_instance` argument is not allowed in CLI: `The following argument is not allowed in CLI: 'start_instance'.\nIgnoring argument` (line 410)

Adding it could be an asset! We should add a warning when removing the `PYMAPDL_START_INSTANCE` from the environment variables.

Reply:
You are right about having to add a warning.

However, when you use a CLI, you create a process (for instance ``pymapdl start``), run it and die. This part of the code (removing the ``PYMAPDL_START_INSTANCE`` env var) is only running in the ``pymapdl start``, hence it should not affect the global value of the en var.

I mean that:
```terminal
$ export PYMAPDL_START_INSTANCE=True
$ pymapdl start # launching mapdl
$ echo $PYMAPDL_START_INSTANCE
True
```
Changing the env vars in Python does only affect the Python process and subsequent process launched from it. For instance, MAPDL won't inheritate the ``PYMAPDL_START_INSTANCE`` but it does not matter, because it should not use it.

Reply:
Let me know if you need any further clarification!

Reply:
Also, regarding:
> For now, start_instance argument is not allowed in CLI: The following argument is not allowed in CLI: 'start_instance'.\nIgnoring argument (line 410)
> 
> Adding it could be an asset!

The ``start_instance`` argument does not make sense in a CLI which intends to **launch** MAPDL. If we tell to not start MAPDL using ``PYMAPDL_START_INSTANCE`` what should be the behaviour??
So I think it does make sense to not allow that argument in the CLI.

I'm open to discuss this more if you think I'm missing something though.

Reply:
Thanks for your detailed answer.
I fully agree with you!